docs/walkthrough/walkthrough-oauth2-proxy.md

```diff
@@ -0,0 +1,168 @@
+- `extraArgs.whitelist-domain` allows `oauth2-proxy` to authenticate request for all `.127.0.0.1.nip.io` subdomains
+- `extraArgs.cookie-secure` disables CSRF as we are using HTTP (not HTTPS)
+- `extraArgs.cookie-domain` is used to set the cookie for all `.127.0.0.1.nip.io` subdomains. As `oauth2-proxy` is deployed centrally, it will be exposed at `http://auth.127.0.0.1.nip.io` while the dashboard will be exposed at `http://tekton-dashboard.127.0.0.1.nip.io`. The cookie needs to be available to both urls
+
+Although not used in this walkthrough, Github provider supports restricting logins to members of an organisation, members of a team, or collaborator of a repository for more security.
+
+See supported [configuration parameters](https://oauth2-proxy.github.io/oauth2-proxy/configuration) for the full list of confiuration options.
+
+`oauth2-proxy` also supports other providers, see the list of [supported providers](https://oauth2-proxy.github.io/oauth2-proxy/auth-configuration) to setup a different provider than Github.
+
+## Setting up Dashboard Ingress rule for authentication
+
+With `oauth2-proxy` deployed, you will now secure the Tekton Dashboard `Ingress`.
+
+To do so, you need to add [annotations on the ingress](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md#external-authentication) to let `NGINX ingress controller` know that the ingress requires authentication and how to authenticate.
+
+Two annotations are used here:
+- `nginx.ingress.kubernetes.io/auth-url` indicates that the ingress uses an external service that provides authentication. This will point to `oauth2-proxy` service
+- `nginx.ingress.kubernetes.io/auth-signin` indicates the url used to sign in when the user is not logged in. This will point to `oauth2-proxy` sign in endpoint, setting the desired redirect url in the query parameters so that users are redirected to the Tekton Dashboard after logging in
+
+To apply the ingress annotations run the command below:
+
+```bash
+cat <<EOF | kubectl apply -n tekton-pipelines -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: tekton-dashboard
+  annotations:
+    nginx.ingress.kubernetes.io/auth-url: "http://oauth2-proxy.tools.svc.cluster.local/oauth2/auth"
+    nginx.ingress.kubernetes.io/auth-signin: "http://auth.127.0.0.1.nip.io/oauth2/sign_in?rd=http://\$host\$request_uri"
+spec:
+  rules:
+  - host: tekton-dashboard.127.0.0.1.nip.io
+    http:
+      paths:
+      - backend:
+          serviceName: tekton-dashboard
+          servicePort: 9097
+EOF
+```
+
+**NOTE:** The url set in the `nginx.ingress.kubernetes.io/auth-url` is the in-cluster dns name of the `oauth2-proxy` service. We can not use `auth.127.0.0.1.nip.io` here because it resolves to `127.0.0.1` and therefore the ingress controller would call itself instead of the `oauth2-proxy` service.
```

Comment:
```suggestion
**NOTE:** The URL set in the `nginx.ingress.kubernetes.io/auth-url` is the in-cluster DNS name of the `oauth2-proxy` service. We can't use `auth.127.0.0.1.nip.io` here because it resolves to `127.0.0.1` and therefore the ingress controller would call itself instead of the `oauth2-proxy` service.
```